tests/test_widgets.py

```diff
@@ -1832,6 +1834,105 @@ def testExecution(self):
         self.presenter.importTomography.assert_called_with('dummypath', [1.0, 2.0, 3.0], [0.0, 1.0, 2.0])
 
 
+class TestCurrentCoordinatesDialog(unittest.TestCase):
+    @mock.patch("sscanss.app.window.presenter.MainWindowModel", autospec=True)
+    def setUp(self, model_mock):
+        self.view = TestView()
+        self.mock_instrument = mock.create_autospec(Instrument)
+        self.mock_instrument.positioning_stack = self.createPositioningStack()
+
+        self.model_mock = model_mock
+        self.model_mock.return_value.instruments = ['dummy']
+        self.model_mock.return_value.instrument = self.mock_instrument
+        self.model_mock.return_value.fiducials_changed = TestSignal()
+        self.model_mock.return_value.instrument_controlled = TestSignal()
+
+        points = np.rec.array([([0.0, 1.0, 2.0], False), ([2.0, 0.0, 1.0], True), ([0.0, 1.0, 1.0], True)],
+                              dtype=POINT_DTYPE)
+        self.model_mock.return_value.fiducials = points
+        self.presenter = MainWindowPresenter(self.view)
+        self.view.scenes = mock.create_autospec(SceneManager)
+        self.view.presenter = self.presenter
+        self.model_mock.return_value.alignment = None
+        self.dialog = CurrentCoordinatesDialog(self.view)
+
+    @staticmethod
+    def createPositioner():
+        q1 = Link("X", [1.0, 0.0, 0.0], [0.0, 0.0, 0.0], Link.Type.Prismatic, -200.0, 200.0, 0)
+        return SerialManipulator("", [q1])
+
+    @staticmethod
+    def createPositioningStack():
+        y_axis = create_cuboid(200, 10, 200)
+        z_axis = create_cylinder(25, 50)
+        q1 = Link("Z", [0.0, 0.0, 1.0], [0.0, 0.0, 0.0], Link.Type.Revolute, -3.14, 3.14, 0, z_axis)
+        q2 = Link("Y", [0.0, 1.0, 0.0], [0.0, 0.0, 0.0], Link.Type.Prismatic, -200.0, 200.0, 0, y_axis)
+        s = SerialManipulator("", [q1, q2], custom_order=[1, 0], base=Matrix44.fromTranslation([0.0, 0.0, 50.0]))
+        return PositioningStack(s.name, s)
+
+    @mock.patch('sscanss.app.dialogs.misc.np.savetxt', autospec=True, return_value=True)
+    @mock.patch('sscanss.app.dialogs.misc.QtWidgets.QFileDialog.getSaveFileName', return_value=('dummy', None))
+    def testCurrentCoordinatesDialog(self, dialog_mock, savetxt):
+        savetxt.assert_not_called()
+        # Test non aligned case that fiducials are parsed unchanged
+        self.assertEqual(self.model_mock.return_value.fiducials['points'][1, 1],
```

Comment:
You should not `self.assertEqual` to compare float most times this a problem waiting to happen use `self.assertAlmostEqual`

Reply:
I have changed this in all but lines 1878 and 1880 where the value should just be a pointer (I think)

Reply:
Don't forget specify the appropriate number of decimals to check